### DISCUSSION TITLE
with-editor not show (commit hangs) after pre-commit install

### DISCUSSION THREAD

**Discussion by tshu-w** [magit/magit#4556] [Support] [answered]
1. what behavior you expected: magit works as normal after [pre-commit](https://pre-commit.com) install hook
2. what behavior you observed: with-editor not show (commit hangs forever)
3. and how we can reproduce the issue: 
    1. install pre-commit
    2. general pre-commit sample config `pre-commit sample-config > .pre-commit-config.yaml`
    3. install commit hook `pre-commit install`

Not error.

Magit 1eb183e, Git 2.17.1, Emacs 29.0.50, darwin

How can I debug?

**Comment by tarsius** [comment, marked as answer]
Magit doesn't run `git commit ...` in an interactive shell so if a hook asks the user for input, then the user never sees that and the hook forever waits for it. You can explicitly tell magit about known prompts by adding an entry to `magit-process-prompt-functions` (and `magit-process-password-prompt-regexps` and `magit-process-username-prompt-regexps`).

**Reply by tshu-w** [reply]
These hooks do not accept input. If I press $ to see the output, I can see that the contents of the commit hook have been executed successfully

**Reply by tshu-w** [reply]
BTW, this issue only happen on tramp.
